### PR TITLE
Add a few unit tests

### PR DESCRIPTION
**Pull request by DLochmelis33**

### CODE REVIEW

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/BarrierTest.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            it.x = 1
+            it.flush = it.x + 1
+        }
+        val f2 = t2.start(ctx) {
+            sleep(1u)
+            assertNotEquals(1, it.x)
+            it.barrier.await()
+            sleep(1u)
+            assertEquals(1, it.x)
+        }
```
Comment (eupp) [inline core/src/nativeTest/kotlin/org/jetbrains/litmuskt/BarrierTest.kt:47]: wouldn't it be easier to have a test like this?
```kotlin

class Context(
      val barrier: Barrier,
      var x: Int = 0,
      @Volatile
      var flag1: Boolean = false // ensure that changes are visible
      @Volatile
      var flag2: Boolean = false // ensure that changes are visible
  )

val f1 = t1.start(ctx) {
    it.x += 1
    it.flag1 = true
    it.barrier.await()
    assert(it.flag2 == true) // volatile flag read also induces synchronize-with relation  
    assert(it.x == 2)
}
val f2 = t2.start(ctx) {
    it.x += 1
    it.flag2 = true
    it.barrier.await()
    assert(it.flag1 == true) // volatile flag read also induces synchronize-with relation  
    assert(it.x == 2)
}
```

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/IntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            r2 = x
+        }
+        spec {
+            accept(0, 2)
+            accept(1, 2)
+            interesting(0, 1)
+            interesting(1, 1)
+
+            interesting(0, 0)
+            interesting(1, 0)
```
Comment (eupp) [inline core/src/nativeTest/kotlin/org/jetbrains/litmuskt/IntegrationTest.kt]: aren't `(1, 1)`, `(0, 0)`, and `(1, 0)` actually forbidden?

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/ThreadlikeTest.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    private fun testThreadlike(t: Threadlike) {
+        class IntHolder(var x: Int)
+
+        val holder = IntHolder(5)
+        val future = t.start(holder) {
+            it.x = 7
+        }
+        future.await()
+        assertEquals(7, holder.x)
```
Comment (eupp) [inline core/src/nativeTest/kotlin/org/jetbrains/litmuskt/ThreadlikeTest.kt]: why `5` and `7` ?)
why not just `0` and `1`, or even `false` and `true` ?

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/BarrierTest.kt (hunk 1 above)
Comment (DLochmelis33) [reply core/src/nativeTest/kotlin/org/jetbrains/litmuskt/BarrierTest.kt:47]: I want to check that one thread is actually waiting for the other. Still, I extended the test a little and added some comments.

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/IntegrationTest.kt (hunk 2 above)
Comment (DLochmelis33) [reply core/src/nativeTest/kotlin/org/jetbrains/litmuskt/IntegrationTest.kt]: No idea :zany_face: it's not like we have a memory model after all. Now I'm curious, I ran it on Arm and these outcomes did not appear, so I guess they are indeed forbidden?

File: core/src/nativeTest/kotlin/org/jetbrains/litmuskt/ThreadlikeTest.kt (hunk 3 above)
Comment (DLochmelis33) [reply core/src/nativeTest/kotlin/org/jetbrains/litmuskt/ThreadlikeTest.kt]: The first number that comes to my mind is usually `5`, the next is usually either `3` or `7`... Anyway, changed to `0` and `1`.

**Pull request merged**